src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -2687,4 +2670,304 @@ instruct vblend(vReg dst, vReg src1, vReg src2, vRegMask_V0 v0) %{
+  format %{ "vcvtFtoX_narrow $dst, $src\t# KILL $tmp" %}
+  ins_encode %{
+    __ rvv_vsetvli(T_FLOAT, Matcher::vector_length_in_bytes(this, $src));
+    __ vfcvt_x_f_v(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg));
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    __ vector_integer_narrow(as_VectorRegister($dst$$reg), bt, Matcher::vector_length_in_bytes(this),
+                             as_VectorRegister($dst$$reg), T_INT, as_VectorRegister($tmp$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vcvtFtoI(vReg dst, vReg src) %{
+  predicate(Matcher::vector_element_basic_type(n) == T_INT ||
+            Matcher::vector_element_basic_type(n) == T_LONG);
+  match(Set dst (VectorCastF2X src));
+  format %{ "vcvtFtoX $dst, $src" %}
+  ins_encode %{
+    __ rvv_vsetvli(T_FLOAT, Matcher::vector_length_in_bytes(this, $src));
+    __ vfcvt_x_f_v(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg));
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    if (bt == T_LONG) {
+      __ vector_integer_extend(as_VectorRegister($dst$$reg), T_LONG,
+                               Matcher::vector_length_in_bytes(this), as_VectorRegister($dst$$reg), T_INT);
+    }
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vcvtFtoD(vReg dst, vReg src) %{
+  predicate(Matcher::vector_element_basic_type(n) == T_DOUBLE);
+  match(Set dst (VectorCastF2X src));
+  effect(TEMP_DEF dst);
+  format %{ "vcvtFtoD $dst, $src" %}
+  ins_encode %{
+    __ rvv_vsetvli(T_FLOAT, Matcher::vector_length_in_bytes(this, $src), Assembler::mf2);
+    __ vfwcvt_f_f_v(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+// VectorCastD2X
+
+instruct vcvtDtoI(vReg dst, vReg src, vReg tmp) %{
+  predicate(Matcher::vector_element_basic_type(n) == T_BYTE ||
+            Matcher::vector_element_basic_type(n) == T_SHORT ||
+            Matcher::vector_element_basic_type(n) == T_INT);
+  match(Set dst (VectorCastD2X src));
+  effect(TEMP_DEF dst,TEMP tmp);
+  format %{ "vcvtDtoI $dst, $src\t# KILL $tmp" %}
+  ins_encode %{
+    __ rvv_vsetvli(T_DOUBLE, Matcher::vector_length_in_bytes(this, $src));
+    __ vfcvt_x_f_v(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg));
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    __ vector_integer_narrow(as_VectorRegister($dst$$reg), bt, Matcher::vector_length_in_bytes(this),
+                             as_VectorRegister($dst$$reg), T_LONG, as_VectorRegister($tmp$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vcvtDtoL(vReg dst, vReg src) %{
+  predicate(Matcher::vector_element_basic_type(n) == T_LONG);
+  match(Set dst (VectorCastD2X src));
+  format %{ "vcvtDtoL $dst, $src" %}
+  ins_encode %{
+    __ rvv_vsetvli(T_LONG, Matcher::vector_length_in_bytes(this));
+    __ vfcvt_x_f_v(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vcvtDtoF(vReg dst_src1, vReg tmp) %{
```

Comment:
Why not break down 'dst_src1' into two seperate 'dst' and 'src' inputs like you do for 'vcvtFtoD' ?

Reply:
Fixed.